parser.py

```diff
@@ -483,58 +489,82 @@ def parse_direct_messages(data_folder, username, users, user_id_URL_template, dm
-            markdown += f'## Conversation between {user1_handle} and {user2_handle}: ##\n'
+            user1_id, user2_id = conversation_id.split('-')
             messages = []
             if 'messages' in dm_conversation:
                 for message in dm_conversation['messages']:
                     if 'messageCreate' in message:
-                        messageCreate = message['messageCreate']
-                        if all(tag in messageCreate for tag in ['senderId', 'recipientId', 'text', 'createdAt']):
-                            from_id = messageCreate['senderId']
-                            to_id = messageCreate['recipientId']
-                            body = messageCreate['text']
-                            created_at = messageCreate['createdAt'] # example: 2022-01-27T15:58:52.744Z
-                            timestamp = int(round(datetime.datetime.strptime(created_at, '%Y-%m-%dT%X.%fZ').timestamp()))
-                            from_handle = users[from_id].handle if from_id in users else user_id_URL_template.format(from_id)
-                            to_handle = users[to_id].handle if to_id in users else user_id_URL_template.format(to_id)
-                            message_markdown = f'\n\n### {from_handle} -> {to_handle}: ({created_at}) ###\n```\n{body}\n```'
+                        message_create = message['messageCreate']
+                        if all(tag in message_create for tag in ['senderId', 'recipientId', 'text', 'createdAt']):
+                            from_id = message_create['senderId']
+                            to_id = message_create['recipientId']
+                            body = message_create['text']
+                            created_at = message_create['createdAt']  # example: 2022-01-27T15:58:52.744Z
+                            timestamp = \
+                                int(round(datetime.datetime.strptime(created_at, '%Y-%m-%dT%X.%fZ').timestamp()))
+                            from_handle = users[from_id].handle if from_id in users \
+                                else user_id_url_template.format(from_id)
+                            to_handle = users[to_id].handle if to_id in users else user_id_url_template.format(to_id)
+                            message_markdown = f'\n\n### {from_handle} -> {to_handle}: ' \
+                                               f'({created_at}) ###\n```\n{body}\n```'
                             messages.append((timestamp, message_markdown))
-            messages.sort(key=lambda tup: tup[0])
+
+            # find identifier for the conversation
+            other_user_id = user2_id if (user1_id in users and users[user1_id].handle == username) else user1_id
+
+            # collect messages per identifying user in conversations_messages dict
+            for message in messages:
+                conversations_messages[other_user_id].append(message)
```

Comment:
Instead of this for-loop:

`conversations_messages[other_user_id].extend(messages)`

Reply:
ok, done :)